apps/code_rag.py

```diff
@@ -0,0 +1,208 @@
+"""
+Code RAG example using AST-aware chunking for optimal code understanding.
+Specialized for code repositories with automatic language detection and 
+optimized chunking parameters.
+"""
+
+import sys
+from pathlib import Path
+
+# Add parent directory to path for imports
+sys.path.insert(0, str(Path(__file__).parent))
+
+from base_rag_example import BaseRAGExample, create_text_chunks
+from llama_index.core import SimpleDirectoryReader
+
```

Comment:
```suggestion
from .utils import CODE_EXTENSIONS
```

Reply:
This wouldn't work as we are not in the same directory no? We've got an __init__.py in chunking/  so `chunking` should be correct (or `chunking.utils`, but not `.utils`)
I've left it as is